lib/cli.rb

```diff
@@ -0,0 +1,164 @@
+require_relative '../config/environment'
+require 'pry'
+require 'colorize'
+
+
+class CLI
+
+  
+
+  def create_user
+    puts `clear`
+    puts "Please type in your name...then press enter.. I guess..."
+    user_name = STDIN.gets.chomp.strip 
+    $user = User.find_or_create_by(name: user_name)
+  end 
+  
+  def greeting
+    puts `clear`
+    puts "Weather is breezy".colorize(:light_magenta)
+    puts 
+    puts "Weather is cool" 
+    puts 
+    puts "Don't get "+"washed".colorize(:light_magenta).italic+" away by too many choices."
+    puts 
+    puts "~ Select an option below "+"#{$user.name}".colorize(:cyan).italic+" ~"
+    puts 
+  end 
+
+  def cli_program
+    is_running = true 
+    while is_running 
+      puts "1. Search "+ "Weather".colorize(:cyan)
+      puts "2. View Search History"
+      puts "3. Delete History"
+      puts "4. Change a Search Record"
+      puts "5. Exit program"
+
+      choice = STDIN.gets.chomp 
+
+      if choice == "1"
+        puts `clear`
+        Search.all.each do |search|
+          puts "Location: #{search.search_location}"
+        end 
+        puts
+        puts "Please type in a City to see the "+"Weather".colorize(:cyan)+", then press enter" 
+        user_input = STDIN.gets.chomp.strip
+        
+        actual_location = Search.all.find_by(search_location: user_input)
+        
+
+          if actual_location
+            weather_description =  Weather.find_by(search_id: actual_location.id)
+            puts "The "+"Weather".colorize(:cyan)+" in #{user_input} is "+"#{weather_description.description}".colorize(:cyan)
+            puts
+            puts "Wasn't expecting that weather, now were you."
+            puts 
+            saved_location = Search.create(search_location: user_input)
+
+            saved_location_in_history = UsersSearch.create(search_id: saved_location.id, user_id: $user.id)
+            
+          else
+            is_running = true            
+            puts
+            puts `clear`
+            puts "That City doesn't have "+"weather.".colorize(:cyan)+" WIERD. Please re-select Search "+"Weather.".colorize(:cyan) 
+          end 
+            
+          
+      elsif choice == "2" 
+        puts `clear`
+        puts 
+        UsersSearch.all.each do |history|
+          puts "Name - #{history.user.name} and Location - #{history.search.search_location}" 
+        puts 
+        end 
+        puts "Someone's being "+"drafty".colorize(:cyan)+"..."
+        puts 
+
+        # writing out my idea 
```

Comment:
You know my shpiel on comments, but it's your choice in the end. If you think this adds context which is useful to someone else coming into the code leave it. But if it was a personal comment for yourself, I would recommend deleting it.